common/app/routes/challenges/components/classic/Classic.jsx

```diff
@@ -88,8 +100,26 @@ export class Challenge extends PureComponent {
       file,
       mode,
       showPreview,
-      executeChallenge
+      executeChallenge,
+      submitChallenge,
+      isChallengePassed,
+      testsInvalid
     } = this.props;
+
+    // A map of 'keydown' events
+    let map = {};
+    let keyBindings = (event) => {
```

Comment:
Do not create function props in render methods. Use and instance method instead.